.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,56 @@
+# Build and push a docker image whenever toolchain script is updated. It is
+# important that this workflow always runs to be able to trigger the default
+# branch ci check.
+name: Docker Build
+
+on:
+  push:
+    branches:
+      - trunk
```

Comment:
Can't you use `on.push.paths` here to simplify the checking?

Reply:
It is a little complicated than that :) On the default branch we also want to execute `ci` to display the build badge. Currently we run ci check after the image is built and pushed so that we don't have to build it both for the `docker-build` and `ci`.
The `ci` workflow is triggered on PRs and pushes except for the default branch and also when the ci is finished for `trunk`. Because of this `docker-build` should always run to trigger `ci` on `trunk`. This also guarantees `trunk`'s `ci` is actually run on the live image just in case.
I know this flow is a little counterintuitive but upon much trial and error I decided that this is the way to go.

Reply:
Hmm I think I have found a better solution now. I will push that instead. Let me verify that first.

Reply:
Unfortunately it is not possible to prevent triggering a workflow if a path match occurs. i.e;
```
on:
  push:
    paths:
      - "tools/build-toolchain.sh"
      - "Dockerfile"
```
will trigger if relevant files change but there is no "else" to it that we can start `ci` without building an image.